File: TypeScript/codewars/Typescript/Take-a-Ten-Minutes-Walk.ts
export function isValidWalk(walk: string[]): boolean {
	if (walk.length !== 10) return false;

	let n = 0;
	let s = 0;
	let e = 0;
	let w = 0;

	for (const direction of walk) {
		if (direction === "n") n++;
		if (direction === "s") s++;
		if (direction === "e") e++;
		if (direction === "w") w++;
	}

	return n === s && e === w;
}

File: Csharp/CodeWars/src/Completed/TakeaTenMinutesWalk.cs
namespace CodeWars.Completed;

public static class TakeaTenMinutesWalk
{
    public static bool IsValidWalk(string[] walk)
    {
        if (walk.Length != 10) return false;

        int n = 0;
        int s = 0;
        int e = 0;
        int w = 0;

        foreach (var direction in walk)
        {
            if (direction == "n") n++;
            if (direction == "s") s++;
            if (direction == "e") e++;
            if (direction == "w") w++;
        }

        return n == s && e == w;
    }
}
